homeassistant/components/recorder/statistics.py

```diff
@@ -485,6 +485,11 @@ def compile_statistics(instance: Recorder, start: datetime, fire_events: bool) -
 
     The actual calculation is delegated to the platforms.
     """
+    # Define modified_statistic_ids outside of the "with" statement as context managers
+    # with exception suppressing prevent defining the value of the local variable, yet
+    # the flow continues
+    modified_statistic_ids = None
```

Comment:
I think we should instead indent `if modified_statistic_ids:` so its within the context manager

Reply:
Makes sense indeed!

Just like in `compile_missing_statistics()`.

Reply:
Done.